tests/unit/test_config.py

```diff
@@ -455,3 +455,38 @@ def _test(cp_tz, utc_mode, expected, expected_warnings=0):
     ]
     for case in tests:
         _test(**case)
+
+
+@pytest.mark.parametrize(
+    'cfg_scheduling, valid',
+    [
+        ({'cycling mode': 'integer', 'runahead limit': 'P14'}, True),
+        ({'cycling mode': 'gregorian', 'runahead limit': 'P14'}, True),
+        ({'cycling mode': 'gregorian', 'runahead limit': 'PT12H'}, True),
+        ({'cycling mode': 'gregorian', 'runahead limit': 'P7D'}, True),
+        ({'cycling mode': 'gregorian', 'runahead limit': 'P2W'}, True),
+        ({'cycling mode': 'gregorian', 'runahead limit': '4'}, True),
+
+        ({'cycling mode': 'integer', 'runahead limit': 'PT12H'}, False),
+        ({'cycling mode': 'integer', 'runahead limit': 'P7D'}, False),
+        ({'cycling mode': 'integer', 'runahead limit': '4'}, False),
+        ({'cycling mode': 'gregorian', 'runahead limit': ''}, False),
```

Comment:
It is not possible to set `runahead limit` to null

Reply:
Sensible, an infinite runahead limit is a whole other kettle of fish!